packages/world/src/modules/std-delegations/CallboundDelegationControl.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { DelegationControl } from "../../DelegationControl.sol";
+import { CallboundDelegations } from "./tables/CallboundDelegations.sol";
+
+contract CallboundDelegationControl is DelegationControl {
+  /**
+   * Verify a delegation by checking if the delegator has any available calls left in the CallboundDelegations table and decrementing the available calls if so.
+   */
+  function verify(address delegator, bytes32 resourceSelector, bytes memory funcSelectorAndArgs) public returns (bool) {
+    bytes32 funcSelectorAndArgsHash = keccak256(funcSelectorAndArgs);
+
+    // Get the number of available calls for the given delegator, resourceSelector and funcSelectorAndArgs
+    uint256 availableCalls = CallboundDelegations.get({
+      delegator: delegator,
+      delegatee: _msgSender(),
+      resourceSelector: resourceSelector,
+      funcSelectorAndArgsHash: funcSelectorAndArgsHash
+    });
+
+    if (availableCalls == 1) {
+      // Remove the delegation from the CallboundDelegations table
+      CallboundDelegations.deleteRecord({
+        delegator: delegator,
+        delegatee: _msgSender(),
+        resourceSelector: resourceSelector,
+        funcSelectorAndArgsHash: funcSelectorAndArgsHash
+      });
+      return true;
+    }
+
+    if (availableCalls > 0) {
+      // Decrement the number of available calls
+      unchecked {
```

Comment:
I don't know the consequences of `unchecked` but is it safe for us to wrap this whole statement vs just the decrement?

Reply:
I don't think it would

Reply:
moved it outside of the block, gas didn't change so seems like unchecked doesn't impact the nested calls, but fells still cleaner to have it explicit